### PR TITLE
BackupScheduleSetting: improve BackupScheduled view using useNextBackupSchedule hook

### PR DESCRIPTION
**Pull request by Initsogar**
Related to https://github.com/Automattic/jetpack-backup-team/issues/715

This PR improves the precision of the next backup time estimation by leveraging the `useNextBackupSchedule` hook introduced in https://github.com/Automattic/wp-calypso/pull/95602 that returns the next scheduled backup date.

## Proposed Changes

* Removed the previous logic that calculated hours until the next backup using the last backup date and replaced it with precise scheduling data from the `useNextBackupSchedule` hook.
  * If there are any minutes left in the calculation, the hours are rounded up to the next hour.

CleanShot 2024-10-23 at 20 56 05@2x

Other minor changes:
* ~Removed `defaultProps` on ActionsButton component, which is now deprecated.~ Update: I was going to remove this, but it seems we have some types issues related to this, so I prefer to address it in a new PR.
* Refactor `useNextBackupSchedule` to named export instead of default export, so we could add more hooks in the same `hooks` file without issues.
* Update schedule backup message in settings page as suggested in p1729507641013709-slack-C07LJAE7AVD

| Scenario | Before | After |
|---|---|---|
| Time set by default | CleanShot 2024-10-23 at 21 36 41@2x | CleanShot 2024-10-23 at 21 37 07@2x |
| Time set by user | CleanShot 2024-10-23 at 21 34 07@2x | CleanShot 2024-10-23 at 21 32 16@2x | 

## Why are these changes being made?

The previous logic considered the last completed backup to calculate a relative date, which is not always accurate, especially since we introduced [on-demand backups](https://jetpack.com/blog/on-demand-backups-at-your-fingertips/).

https://github.com/Automattic/jetpack-backup-team/issues/715

## Testing Instructions

* Spin up a Jetpack Cloud live branch.
* Pick a site with Jetpack VaultPress Backup plan with no backups today.
* Navigate to the `Backup` page.
* You should see the `Backup Scheduled` view and the time should be relative to the site scheduled time.
* You can try updating the daily backup time setting and see how it affects the "In the next % hour/s" message.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Comment by Initsogar** [comment]
@ilonagl here I included the changes on the changes on the scheduled setting message:

| Scenario | Before | After |
|---|---|---|
| Time set by default | CleanShot 2024-10-23 at 21 36 41@2x | CleanShot 2024-10-23 at 21 37 07@2x |
| Time set by user | CleanShot 2024-10-23 at 21 34 07@2x | CleanShot 2024-10-23 at 21 32 16@2x |

**Review by ilonagl — APPROVED** [review]

File: client/components/jetpack/daily-backup-status/status-card/backup-scheduled.jsx [hunk 1]
```diff
@@ -35,16 +36,22 @@ const BackupScheduled = ( { lastBackupDate } ) => {
 
 	const lastBackupTime = lastBackupDate.format( 'LT' );
 
-	// Calculates the remaining hours for the next backup + 3 hours of safety margin
-	const DAY_HOURS = 24;
-	const hoursForNextBackup = DAY_HOURS - today.diff( lastBackupDate, 'hours' ) + 3;
+	const { date: nextBackupDate } = useNextBackupSchedule();
+
+	// Calculate the time difference for hours and minutes
+	const hoursForNextBackup = nextBackupDate.diff( today, 'hours' );
```
Comment (Initsogar) [inline client/components/jetpack/daily-backup-status/status-card/backup-scheduled.jsx]: I was testing this on Jetpack Cloud live branch and noted this error:

image

It seems nextBackupDate could be potentially null, so I'd need to add some guard checks here.

**Review by mavegaf — APPROVED** [review]

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/16942926

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @Initsogar for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16942926).

**Pull request merged**